guidance/_grammar.py

```diff
@@ -603,7 +607,7 @@ def __init__(self):
 class Join(GrammarFunction):
     __slots__ = ("nullable", "values", "name", "hidden", "commit_point", "capture_name", "max_tokens")
```

Comment:
Related to this... I'm seeing quite a few places where `__slots__` contains `"hidden", "commit_point", "capture_name"` ... is there a base class lurking beneath these? Should it be `GrammarFunction` itself?

Reply:
There probably is :) ...however we can address that in another PR. I think we are likely to call these grammar objects something like "GuidanceScript" objects in the future (similar to torchscript), because they will encode more than similar grammar idioms.